### ISSUE TITLE
_mi_heap_main "cookie" and "random" initializations may be swapped

### ISSUE THREAD

**Issue by colesbury** [microsoft/mimalloc#131]
The static initialization of `_mi_heap_main` looks like:

https://github.com/microsoft/mimalloc/blob/cbc826beb598f48099a3b35d6f228103a630085a/src/init.c#L99-L113

From the comment it looks like `_mi_heap_main.cookie` is initialized to `0xCDCDCDCDCDCDCDCDUL`, but that position corresponds to _mi_heap_main.random. Printing `cookie` and `random` at the beginning and end of mi_process_init gives the following (on macOS):

```
// start of mi_process_init 
_mi_heap_main.random = cdcdcdcdcdcdcdcd
_mi_heap_main.cookie = 0

// end of mi_process_init ("random" is random)
_mi_heap_main.random = a75bab6bdb39f6e9
_mi_heap_main.cookie = 0
```

**Comment by daanx** [comment]
Thanks Sam!! You keep finding bugs and I am embarrassed by this one .. Will fix this soon but if you can send a PR against master that would be great as I have little time until next week.

**Comment by daanx** [comment]
Fixed in [dev 7429964]  -- Thanks!
